Explorer/Assets/DCL/Infrastructure/CrdtEcsBridge/JsModulesImplementation/Communications/CommunicationsControllerAPIImplementationBase.cs

```diff
@@ -57,40 +57,56 @@ public void Dispose()
 
-            foreach (PoolableByteArray poolable in broadcastData)
-                if (poolable.Length > 0)
+            foreach (ITypedArray<byte> data in broadcastData)
+                if (data.Length > 0)
                 {
-                    byte firstByte = poolable.Span[0];
-
-                    ISceneCommunicationPipe.ConnectivityAssertiveness assertiveness = firstByte == (int)CommsMessageType.REQ_CRDT_STATE
-                        ? ISceneCommunicationPipe.ConnectivityAssertiveness.DELIVERY_ASSERTED
-                        : ISceneCommunicationPipe.ConnectivityAssertiveness.DROP_IF_NOT_CONNECTED;
-
-                    // Filter CRDT messages before sending
-                    if (firstByte == (int)CommsMessageType.CRDT)
-                    {
-                        Span<byte> filtered = stackalloc byte[poolable.Memory.Span.Length];
-                        int filteredLength = FilterCRDTMessage(poolable.Memory.Span, filtered);
-                        EncodeAndSendMessage(ISceneCommunicationPipe.MsgType.Uint8Array, filtered.Slice(0, filteredLength), assertiveness, recipient);
-                        continue;
-                    }
-
-                    // Filter RES_CRDT_STATE messages before sending
-                    if (firstByte == (int)CommsMessageType.RES_CRDT_STATE)
-                    {
-                        Span<byte> filtered = stackalloc byte[poolable.Memory.Span.Length];
-                        int filteredLength = FilterCRDTStateMessage(poolable.Memory.Span, filtered);
-                        EncodeAndSendMessage(ISceneCommunicationPipe.MsgType.Uint8Array, filtered.Slice(0, filteredLength), assertiveness, recipient);
-                        continue;
-                    }
-
-                    EncodeAndSendMessage(ISceneCommunicationPipe.MsgType.Uint8Array, poolable.Memory.Span, assertiveness, recipient);
+                    ulong length = data.Length;
+                    var instance = this;
+
+                    data.InvokeWithDirectAccess(
+                        static (ptr, args) => {
+                            Span<byte> span;
+                            unsafe
+                            {
+                                span = new Span<byte>(ptr.ToPointer(), (int) args.length);
+                            }
+
+                            byte firstByte = span[0];
+
+                            ISceneCommunicationPipe.ConnectivityAssertiveness assertiveness = firstByte == (int)CommsMessageType.REQ_CRDT_STATE
+                                ? ISceneCommunicationPipe.ConnectivityAssertiveness.DELIVERY_ASSERTED
+                                : ISceneCommunicationPipe.ConnectivityAssertiveness.DROP_IF_NOT_CONNECTED;
+
+                            // Filter CRDT messages before sending
+                            if (firstByte == (int)CommsMessageType.CRDT)
+                            {
+                                Span<byte> filtered = stackalloc byte[span.Length];
```

Comment:
What's the largest possible span length here? How high is the risk of running out of stack?